model/connection.js

```diff
@@ -54,27 +131,106 @@ class Connection {
      */
     constructor(name = Validation.DEFAULT_STR_VALUE, id = Validation.DEFAULT_STR_VALUE, layer = Validation.DEFAULT_STR_VALUE,
                 source = null, sinks = []) {
+        super(name, id);
+        this.layer = layer;
+        this.source = source;
+        this.sinks = sinks;
 
+        // Connection Feature fields
+        this.connection = Validation.DEFAULT_STR_VALUE;
+        this.width = Validation.DEFAULT_DIM_VALUE;
+        this.depth = Validation.DEFAULT_DIM_VALUE;
+        this.sourcePoint = null;
+        this.sinkPoint = null;
+        this.type = Validation.DEFAULT_CON_TYPE;
```

Comment:
Is this a reserved keyword `type`?

Reply:
I didn't get any errors, but I changed it just in case. It's also more descriptive now.